File: file-handling/handling-text-files/python/write_to_text_file.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4
     

def write_to_file():
    """Example of writing to a text file"""
    file_object = open("twinkle.txt", mode="w")  
    file_object.write("Twinkle, twinkle, little star,\n")
    file_object.write("How I wonder what you are!\n")
    file_object.write("Up above the world so high,\n")
    file_object.write("Like a diamond in the sky.\n")
    file_object.close()

    
# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    write_to_file()

File: file-handling/handling-text-files/c-sharp/write_to_text_file.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;

namespace IsaacCodeSamples
{
    class Program
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            WriteToFile();
        }


        // Example of writing to a text file
        public static void WriteToFile() {
            using (StreamWriter writer = new StreamWriter("twinkle.txt", false)) {
                writer.WriteLine("Twinkle, twinkle, little star,");
                writer.WriteLine("How I wonder what you are!");
                writer.WriteLine("Up above the world so high,");
                writer.WriteLine("Like a diamond in the sky.");
            } // The stream is now closed
        }

        
    }
}